package/MDAnalysis/analysis/hbonds/wbridge_analysis.py

```diff
@@ -820,93 +821,108 @@ def _log_parameters(self):
         acceptor names", self.forcefield)
 
     def _update_selection(self):
-        self._s1 = self.u.select_atoms(self.selection1)
-        self._s2 = self.u.select_atoms(self.selection2)
+        self._s1_donors = []
+        self._s1_h_donors = {}
+        self._s1_donors_h = defaultdict(list)
+        self._s1_acceptors = []
 
-        if self.filter_first and self._s1:
+        self._s2_donors = []
+        self._s2_h_donors = {}
+        self._s2_donors_h = defaultdict(list)
+        self._s2_acceptors = []
+
+        self._s1 = self.u.select_atoms(self.selection1).ix
+        self._s2 = self.u.select_atoms(self.selection2).ix
```

Comment:
Here you're turning this selection into indices, but then only using these indices to slice `u.atoms` to get back the AtomGroup you used to have.  I'm not sure why this change was made?

Reply:
As is mentioned in #2177, @orbeckst mentioned that it is better to store the relevant atom as atom index. This is a way of making internal representation consistent.
The more practical reason is that I need a double linked list which links the hydrogen bond donor heavy atom to the hydrogen as the bonded information is not always available. This is currently done using a dictionary. The key of the dictionary must be unmutable and thus, the atom index is chosen as a way of representing the atom. 
To avoid the confusion that in some part of the code, the atom group is represented as atom group and in another part of the code, atom group is represented as a list of indexes. I made the decision that all of the representation of atom group is done using atom index.

Reply:
From what I can tell that issue is mostly about how the results are reported to users, internally you want to use whatever works best.  It doesn't make much sense to me to strip down an AtomGroup to indices only to reconstruct it later.

You can make a dictionary with `Atom` objects as keys.

But ok if this is how you want it and it works then we can go with this.

Reply:
Thank you for the suggestion. I never know that atoms are unmutable, so when I need something unmutable, I turned to atom index. What is being manipulated here are all private attributes, so It is for use within the class only and is not meant to be shown to the user.